build/update-attribution-files/create_pr.sh

```diff
@@ -205,13 +205,16 @@ EOF
 }
 
 function pr::create::attribution() {
-    local -r pr_title="Update ATTRIBUTION.txt files"
+    local pr_title="Update ATTRIBUTION.txt files"
     local commit_message="[PR BOT] Update ATTRIBUTION.txt files"
     local pr_branch="attribution-files-update-$MAIN_BRANCH"
     local -r pr_body=$(pr::create::pr_body "attribution")
     
     local force_push="true"
     if [ -n "${CODEBUILD_RESOLVED_SOURCE_VERSION:-}" ]; then
+        if [ "${CODEBUILD_RESOLVED_SOURCE_VERSION}" != "$MAIN_BRANCH" ]; then
```

Comment:
This variable will give you the commit hash in CodeBuild, not the branch. In addition to CodeBuild, this script is also run in the attribution Prowjobs where we use `PULL_BASE_REF` to get the branch name. When invoking the script through CodeBuild too, we pass in the PULL_BASE_REF as an override, so I believe we can just use `PULL_BASE_REF` here. Or the `CODEBUILD_SOURCE_VERSION` like in other places.

We also need to do this check when defining the checksums PR title.

Reply:
You are right. We override the `MAIN_BRANCH` variable with the `CODEBUILD_SOURCE_VERSION` or `PULL_BASE_REF` value if it is non-empty string on lines 36 and 40. So I think I can directly check if $MAIN_BRANCH  != "main" to update the title.